src/project-config.ts

```diff
@@ -24,8 +24,35 @@ export class ProjectConfig {
   shell: string;
   entrypoints: string[];
 
-  constructor(options: ProjectConfigOptions) {
+  static fromConfigFile(filepath: string): ProjectConfig {
+    try {
+      let configContent = fs.readFileSync(filepath, 'utf-8');
```

Comment:
is there really no way to check a file besides catching an exception? `statSync`?